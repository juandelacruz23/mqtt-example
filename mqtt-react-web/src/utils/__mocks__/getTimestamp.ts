export default jest.fn(() => 1487076708000);
