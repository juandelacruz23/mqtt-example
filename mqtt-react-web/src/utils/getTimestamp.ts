// To simplify mocking with jest
export default function getTimestamp() {
  return Date.now();
}
