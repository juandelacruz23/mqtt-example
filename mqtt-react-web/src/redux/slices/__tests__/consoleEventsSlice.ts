import reducer, { add, clear } from "../consoleEventsSlice";
import getTimestamp from "../../../utils/getTimestamp";
jest.mock("../../../utils/getTimestamp");

describe("slice", () => {
  test("add", () => {
    const fakeValue = "fake value";
    const fakeTimestamp = getTimestamp();
    const action = add({
      value: fakeValue,
      timestamp: fakeTimestamp,
    });

    // actions
    expect(action).toMatchSnapshot();

    // reducer
    const newState = reducer([], action);
    expect(newState).toEqual([action.payload]);
  });
  test("clear", () => {
    const action = clear();

    // actions
    expect(action).toMatchSnapshot();

    //reducer
    const newState = reducer(undefined, action);
    expect(newState).toEqual([]);
  });
});
