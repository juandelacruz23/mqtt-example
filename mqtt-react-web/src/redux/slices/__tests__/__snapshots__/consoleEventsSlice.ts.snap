// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`slice add 1`] = `
Object {
  "payload": Object {
    "timestamp": 1487076708000,
    "value": "fake value",
  },
  "type": "consoleEvents/add",
}
`;

exports[`slice clear 1`] = `
Object {
  "payload": undefined,
  "type": "consoleEvents/clear",
}
`;
